src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -178,6 +208,20 @@ ProfileEvents::Counters * ThreadStatus::attachProfileCountersScope(ProfileEvents
     return prev_counters;
 }
 
+void ThreadStatus::TimePoint::SetUp(std::chrono::time_point<std::chrono::system_clock> now)
+{
+    // query_start_time_{microseconds, nanoseconds} are all constructed from the same time point
+    // to ensure that they are all equal up to the precision of a second.
+    nanoseconds = timeInNanoseconds(now);
```

Comment:
Now I think: do we really need 3 separate fields?
Probably, just store everything in nanoseconds and convert on fly.
E.g. like it is done in Stapwatch.